src/solc_0.7/Game/GameMinter.sol

```diff
@@ -0,0 +1,92 @@
+    ///////////////////////////////  Libs //////////////////////////////
+    using SafeMath for uint256;
+
+    ///////////////////////////////  Data //////////////////////////////
+
+    GameToken internal immutable _gameToken;
+    uint256 internal immutable _gameMintingFee;
+    uint256 internal immutable _gameUpdateFee;
+    address internal immutable _feeBeneficiary;
+    IERC20 internal immutable _sand;
+
+    ///////////////////////////////  Functions /////////////////////////
+
+    constructor(
+        GameToken gameTokenContract,
+        address metaTransactionContract,
+        uint256 gameMintingFee,
+        uint256 gameUpdateFee,
+        address feeBeneficiary,
+        IERC20 sand
+    ) {
+        _gameToken = gameTokenContract;
+        _setMetaTransactionProcessor(metaTransactionContract, METATX_SANDBOX);
+        _gameMintingFee = gameMintingFee;
+        _gameUpdateFee = gameUpdateFee;
+        _feeBeneficiary = feeBeneficiary;
+        _sand = sand;
+    }
+
+    /// @notice Function to create a new GAME token
+    /// @param from The address of the one creating the game, included in the gameId
+    /// @param to The address who will be assigned ownership of this game
+    /// @param creation The struct containing ids & ammounts of assets to add to this game,
+    /// along with the uri to set.
+    /// @param editor The address to allow to edit (can also be set later).
+    /// @param subId A random id created on the backend.
+    /// @return gameId The id of the new GAME token (erc721)
+    function createGame(
+        address from,
+        address to,
+        GameToken.Update calldata creation,
+        address editor,
+        uint64 subId
+    ) external override returns (uint256 gameId) {
+        require(msg.sender == from || _isValidMetaTx(from), "CREATE_ACCESS_DENIED");
+        _chargeSand(from, _gameMintingFee);
+        return _gameToken.createGame(from, to, creation, editor, subId);
+    }
+
+    /// @notice Update an existing GAME token.This actually burns old token
+    /// and mints new token with same basId & incremented version.
+    /// @param from The one updating the GAME token.
+    /// @param gameId The current id of the GAME token.
+    /// @param update The values to use for the update.
+    /// @return newId The new gameId.
+    function updateGame(
+        address from,
+        uint256 gameId,
+        GameToken.Update memory update
+    ) external override returns (uint256 newId) {
+        _checkAuthorization(from, gameId);
+        _chargeSand(from, _gameUpdateFee);
+        return _gameToken.updateGame(from, gameId, update);
+    }
+
+    function _chargeSand(address from, uint256 sandFee) internal {
+        if (_feeBeneficiary != address(0) && sandFee != 0) {
+            _sand.transferFrom(from, _feeBeneficiary, sandFee);
+        }
+    }
+
+    function _checkAuthorization(address from, uint256 id) internal view {
+        address gameOwner = _gameToken.ownerOf(id);
```

Comment:
Looking at this today, but currently, a tokenId with a future-version (valid base-id, but a yet-unminted version number)  may actually pass this checkpoint. 
`ownerOf` will return the current owner, even for all possible future versions of a token.
I think only the current version of a token should return the owner. 
We already have the version in storage (as well as in the tokenId), so easy enough to check.

Reply:
You are right, as per erc721, ownerOf must reject if tokenId do not exist

Reply:
We might need to rethink how _ownerOf of ERC721BaseToken.sol is implemented, or at least rename it ?

Reply:
We should also add test for erc721 to ensure this potental bug, like performing bitwise operation on existing tokenId to ensure it throw for non-existing tokens

Reply:
Will mark this resolved, but have opened a Jira Issue to track it. It should be fixed in a separate PR.
https://sandboxgame.atlassian.net/browse/TSBBLOC-180